### DISCUSSION TITLE
How to get the energy deposition of the electron beam in a material?

### DISCUSSION THREAD

**Discussion by CQUswl** [nrc-cnrc/EGSnrc#1107] [General]
I would like to get the energy deposition of the electron beam in some kind of target. For which of the existing EGSnrc GUIs I have not found one that can over do it.  DOSXYZnrc simulation calculates for the photon beam dose, and the other programs do not make it clear. So, I wonder if this can be achieved with other features of EGSnrc, such as egs++. The crux of this question is how I should record the deposition of electron energy in a body mould.

**Comment by CQUswl** [comment]
> I would like to get the energy deposition of the electron beam in some kind of target. For which of the existing EGSnrc GUIs I have not found one that can over do it. DOSXYZnrc simulation calculates for the photon beam dose, and the other programs do not make it clear. So, I wonder if this can be achieved with other features of EGSnrc, such as egs++. The crux of this question is how I should record the deposition of electron energy in a body mould.

![image](https://github.com/nrc-cnrc/EGSnrc/assets/114200268/f85e2a5b-f330-470b-a98e-34bd36d6d933)
Are there any of these options that will fulfil my needs?

**Reply by rtownson** [reply]
`egs_dose_scoring` will give you the energy deposited to all regions in your geometry, or to a list of regions that you provide. Using egs++ is the direction I would recommend. You could also get the dose deposited for an electron beam in a voxel grid using DOSXYZnrc, but it is restricted to only XYZ grids. In egs++ there is much more flexibility in terms of geometry definitions.

If you're new to egs++, consider working through the relevant sections of the [getting started manual.](https://nrc-cnrc.github.io/EGSnrc/doc/getting-started.pdf)

**Reply by CQUswl** [reply]
Thank you very much for your answer! I had previously used DOSXYZnrc to calculate the energy deposition of electrons in the target, but I found that I could not tell whether it was the energy deposition of electrons or photons. Is there an option to distinguish between photons and electrons? Does egs++ have the same problem?

**Reply by rtownson** [reply]
No, unfortunately there's not built-in functionality in egs++ for distinction in the energy deposited between primary and secondary contributions. In any EGSnrc application, note that you can turn off photon production by increasing AP.

**Reply by CQUswl** [reply]
Okay, I got it. Thanks.
